### PR TITLE
Fix bailoutinfo for orphanedcalls where all argout instructions are deadcoded

### PR DESCRIPTION
**Pull request by meg-gupta**
A BailOnNoProfile can be present between StartCall instruction and argouts.
In this case we can recognize orphaned StartCall during filling bailoutinfo in LinearScan only by looking at the opcode.
In case of non orphaned calls, we expect LoweredStartCall.
This change reverts the previous attempt of recognizing such calls by looking at liveargcount and uses opcode instead.

OS#14945178

### CODE REVIEW

File: lib/Backend/LinearScan.cpp [hunk 1]
```diff
@@ -1775,8 +1775,12 @@ LinearScan::FillBailOutRecord(IR::Instr * instr)
             uint outParamCount = bailOutInfo->GetStartCallOutParamCount(i);
             startCallOutParamCounts[i] = outParamCount;
 #ifdef _M_IX86
-            uint orphanedArgCount = 0;
-            uint deadArgCount = 0;
+            if (bailOutInfo->startCallInfo[i].instr->m_opcode == Js::OpCode::StartCall)
+            {
+                // Deadcode might have eliminated the argouts and the call instruction due to a BailOnNoProfile after StartCall
+                // In such cases, StartCall opcode is not changed to LoweredStartCall, mark the StartCall instruction accordingly
+                bailOutInfo->startCallInfo[i].isOrphanedCall = true;
```
Comment (rajatd) [inline lib/Backend/LinearScan.cpp]: Do we even need the bool `isOrphanedCall` on `StartCallInfo` then?

Comment (meg-gupta) [reply lib/Backend/LinearScan.cpp]: We use it later on in Lowerer::EmitEHBailoutStackRestore.

**Comment by rajatd** [comment]
Can we use the instr field there or is it too late then?

Sent from mobile device
________________________________

**Comment by rajatd** [comment]
My comment is really just a cleanup suggestion. Feel free to merge if you think the suggested cleanup can't be done.

**Comment by meg-gupta** [comment]
Peeps happens before EHPatchBailout. Peeps deletes these instructions. Thanks for the review.

**Pull request merged**